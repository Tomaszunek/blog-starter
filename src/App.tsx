import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import './App.css';


export namespace App {
  export interface IProps extends RouteComponentProps<void> {
    numbers: any   
  }
}

export default class App extends React.Component<App.IProps> {    
  public render() {
    return (
      <div className="App">
        abc
      </div>
    );
  }
}


