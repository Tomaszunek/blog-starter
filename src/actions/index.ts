export * from './articles';
export * from './products';
export * from './mainpage';
export * from './abilities';
export * from './projects';
export * from './appSystem';

