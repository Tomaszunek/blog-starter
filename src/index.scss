@import 'src/styles/variables.scss';

body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
  background-color: $body-background;
}

.container {
  padding: 0px 100px;
  background-color: $primary-background;

  @media only screen and (min-width: $sm) and (max-width: $lg) {
    padding: 0px 80px;
  }
  @media only screen and (min-width: $xs) and (max-width: $sm) {
    padding: 0px 60px;
  }
  @media only screen and (max-width: $xs) {
    padding: 0px 40px;ab
  }   
}