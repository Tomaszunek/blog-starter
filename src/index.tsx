import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Application } from './router';
import { configureStore } from './store';
import { createBrowserHistory } from 'history';
import registerServiceWorker from './registerServiceWorker';
import { Provider } from 'react-redux';
import { Router, Switch } from 'react-router';
import Navigation from 'src/pages/Navigation';
import Footer from 'src/pages/Footer';
import './index.scss';

const store = configureStore();
const history = createBrowserHistory();

ReactDOM.render(
  <Provider store={store}>    
    <Router history={history}>
      <div>
        <Navigation/>
          <Switch>
            <Application />
          </Switch>
        <Footer/>
      </div>
    </Router>   
  </Provider>,  
  document.getElementById('root') as HTMLElement
);
registerServiceWorker();