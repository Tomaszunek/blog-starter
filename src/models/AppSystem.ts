export interface IAppSystem {
    loading: boolean   
} 