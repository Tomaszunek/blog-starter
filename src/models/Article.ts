

export interface IArticle {
    id: number,
    name: string,
    body: string
}
    
  
export namespace ArticleFiltes {
    export enum Filter {
        SHOW_ALL = 'all',
        SHOW_ACTIVE = 'active',
        SHOW_COMPLETED = 'completed'
    }
}