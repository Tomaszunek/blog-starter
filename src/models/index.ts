export * from './ArticleModel';
export * from './ProductModel';
export * from './ContentCategory';
export * from './MainPageContentModel';
export * from './AppSystem';

