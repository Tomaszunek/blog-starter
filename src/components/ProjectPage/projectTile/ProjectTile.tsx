import * as React from 'react';
import './projectTile.scss';
import { IArticleModel } from '../../../models';
import CategoryTags from '../../../components/template/categoryTags/categoryTags';

export default class ProjectTile extends React.Component<IProjectTileProps, any> {
    constructor(props: any) {
        super(props);       
    }

    public render() {
        const { name, body, type, image, AppContentCategories, slug, articleType, createdAt, updatedAt } = this.props.article;
        const url = `${type}/${slug}`;
        return (
            <a className={"projectTile " + type} href={(type === "project" ? url : undefined)}>
                <img src={"/images/projects/" +  image} alt=""/>
                <div className="categoryTile">
                    <CategoryTags articleType={type} categories={AppContentCategories}/>
                    <span>{name}</span>
                </div>
                {
                    (type === "project") ? (
                        <table>
                            <tbody>
                                <tr>                            
                                    <td>Type: {articleType}</td>
                                    <td>Added: {createdAt}</td>
                                    <td>Last Update: {updatedAt}</td>
                                </tr>
                            </tbody>
                        </table>
                    ) : (
                        <p>{body}</p>
                    )
                }
            </a>
        );
    }
}

interface IProjectTileProps {
    article: IArticleModel
}