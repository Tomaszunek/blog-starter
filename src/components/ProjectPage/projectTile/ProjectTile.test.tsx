import * as React from 'react';
import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';
import ProjectTile from './ProjectTile'
import { IArticleModel } from 'src/models';

describe('ProjectPage', () => {
  describe('Component should render right', () => {
    let app:any;
    const testMatch:any = {
      path: '',
      params: {},
      url: ''
    };
    const project:IArticleModel ={
        id: 1,
        type: "ability",
        body: "body",
        createdAt: new Date(1545167325 *1000),
        image: "images/placeholder/abc.jpg",
        name: "woda",
        slug: "woda",
        updatedAt: new Date(1545167325 *1000),
        articleType: "type"
    }        
    beforeEach(() => {
      app = mount( 
        <ProjectTile article={project}/>
      )
    })
    it('render correctly snapshot component', () => {
      const tree = renderer
      .create(<ProjectTile article={project}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
    })    
    it('should contain project containar', () => {      
      expect(app.find('.projectTile').length).toEqual(1)               
    })
    it('should contain elements', () => {
        expect(app.find('img').length).toEqual(1)
        expect(app.find('.categoryTile').length).toEqual(1)
    })
    it('should have proper props', () => {
      expect(app.instance().props.article).toEqual(project)
    })   
  });
}); 