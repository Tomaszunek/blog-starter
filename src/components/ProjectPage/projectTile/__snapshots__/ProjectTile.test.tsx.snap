// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectPage Component should render right render correctly snapshot component 1`] = `
<a
  className="projectTile ability"
  href={undefined}
>
  <img
    alt=""
    src="/images/projects/images/placeholder/abc.jpg"
  />
  <div
    className="categoryTile"
  >
    <span>
      woda
    </span>
  </div>
  <p>
    body
  </p>
</a>
`;
